core/silkworm/consensus/base/engine.cpp

```diff
@@ -102,13 +102,7 @@ ValidationResult EngineBase::validate_block_header(const BlockHeader& header, co
         return ValidationResult::kGasAboveLimit;
     }
 
-    if (header.gas_limit < 5000) {
-        return ValidationResult::kInvalidGasLimit;
-    }
-
-    // https://github.com/ethereum/go-ethereum/blob/v1.9.25/consensus/ethash/consensus.go#L267
-    // https://eips.ethereum.org/EIPS/eip-1985
-    if (header.gas_limit > INT64_MAX) {
+    if (header.gas_limit < param::kMinGasLimit || header.gas_limit > param::kMaxGasLimit) {
```

Comment:
Please explain me ... EIP-4803 talks about **transaction gas limit** here instead we check for block gas being over limit. It is obvious that overall block gas limit exceeds `kMaxGasLimit` if any of its transactions exceeds the limit but shouldn't the check be performed at transaction level ?

Reply:
Rethinking ... probably previous version was correct already with proper reference EIP-1985
I believe the meaning of 4803 is to early reject transactions at tx-pool level: if it's already in a block is the block gas check which holds (EIP-1985) not EIP-4803

> Backwards Compatibility
While this is a breaking change, no actual effect should be visible.

>Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) it was possible to include transactions with gasPrice = 0 and thus the gasLimit * gasPrice <= accountBalance calculation could have allowed for arbitrarily large values of gasLimit. However, the rule that the transaction list cannot exceed the block gas limit, and the strict rules about how the block gas limit can change, prevented arbitrarily large values of gasLimit to be in the historical state.

Reply:
My bad, I somehow confused block and transaction gas limits.